internal/resources/policy/type/custom/input_schema.go

```diff
@@ -0,0 +1,221 @@
+
+	case unknownRecipe:
+		fmt.Printf("[ERROR]: No valid input recipe block found: minimum one valid input recipe block is required among: %v. Please check the schema.", strings.Join(recipesAllowed[:], `, `))
+	}
+
+	return []interface{}{flattenInputData}
+}
+
+func validateInput(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(specKey)
+	if !ok {
+		return fmt.Errorf("spec: %v is not valid: minimum one valid spec block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("spec data: %v is not valid: minimum one valid spec block is required among: %v", data, strings.Join(recipesAllowed[:], `, `))
+	}
+
+	specData := data[0].(map[string]interface{})
+
+	v, ok := specData[inputKey]
+	if !ok {
+		return fmt.Errorf("input: %v is not valid: minimum one valid input block is required", v)
+	}
+
+	v1, ok := v.([]interface{})
+	if !ok {
+		return fmt.Errorf("type of input block data: %v is not valid", v1)
+	}
+
+	if len(v1) == 0 || v1[0] == nil {
+		return fmt.Errorf("input data: %v is not valid: minimum one valid input block is required", v1)
+	}
+
+	inputData, _ := v1[0].(map[string]interface{})
+	recipesFound := make([]string, 0)
+
+	if v, ok := inputData[reciperesource.TMCBlockNodeportServiceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCBlockNodeportServiceKey)
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCBlockResourcesKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCBlockResourcesKey)
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCBlockRolebindingSubjectsKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCBlockRolebindingSubjectsKey)
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCExternalIPSKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCExternalIPSKey)
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCHTTPSIngressKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCHTTPSIngressKey)
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCRequireLabelsKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			recipesFound = append(recipesFound, reciperesource.TMCRequireLabelsKey)
+		}
+	}
+
+	if len(recipesFound) == 0 {
+		return fmt.Errorf("no valid input recipe block found: minimum one valid input recipe block is required among: %v", strings.Join(recipesAllowed[:], `, `))
+	} else if len(recipesFound) > 1 {
+		return fmt.Errorf("found input recipes: %v are not valid: maximum one valid input recipe block is allowed", strings.Join(recipesFound, `, `))
```

Comment:
Please remove duplication of error messages from both security and custom policies. Also, please check if we can reuse these formats across policies.

This applies to all other standard messaging formats across policies.